code/core/graf.cpp

```diff
@@ -0,0 +1,160 @@
+#include "device\graf.h"
+
+float graf::CalculateWeight(std::pair<float,float> firstVertex, std::pair<float,float> secondVertex)
+{
+    return sqrt((firstVertex.first - secondVertex.first) * (firstVertex.first - secondVertex.first) +
+             (firstVertex.second - secondVertex.second) * (firstVertex.second - secondVertex.second));
+}
+
+void graf::AddVertex(vertex_graf vertex)
+{
+    adjacencyList.push_back(vertex);
+    for(auto i : vertex.edgeWeight)
+    {
+        adjacencyList[i.first].edgeWeight.push_back({adjacencyList.size() - 1, i.second});
+    }
+}
+
+void graf::DeleteVertex(int index)
+{
+    for(auto & i : adjacencyList)
+    {
+        int tmp = 0;
+        for(auto & j : i.edgeWeight)

```

Comment:
do you really need range for if you use tmp as variable of cycle. also there are std::find and vector::find

Reply:
In this case, yes. Since we are working with a pair